cmd/juju/caas/add.go

```diff
@@ -363,10 +388,45 @@ var noRecommendedStorageErrMsg = `
 
 // Run is defined on the Command interface.
 func (c *AddCAASCommand) Run(ctx *cmd.Context) (err error) {
+	if c.hostCloudRegion != "" {
+		ctx.Infof("region flag is DEPRECATED. Use 'add-k8s name cloud/region' instead")
+	}
+	if c.cloud != "" {
+		ctx.Infof("cloud flag is DEPRECATED. Use 'add-k8s name cloud/region' instead")
+	}
 	if err := c.verifyName(c.caasName); err != nil {
 		return errors.Trace(err)
 	}
 
+	// validate provided cloud/region
```

Comment:
The cloud and region used here to record where the cluster is running does not have to be known to the controller necessarily. We could add a k8s cluster running via CDK on openstack and that openstack does not have to be managed by the same controller as that which we want to use to run the k8s workloads. The cloud and region are used to ensure juju checks the cluster for opinionated storage choices.